### PR TITLE
feat: events page

### LINKED ISSUE #28: Event page
the information regarding the event page ie , description, link to random chat, link to group chat

### PR DESCRIPTION
**Pull request by zainja**
# Event Page

Resolves #28 

This is displayed when the user clicks on the event card, the user has an option to join a group chat or a random pm chat
## the current page looks on desktop like this
image
## Phone view
image
## URL component
image

### CODE REVIEW

**Review by amishshah — CHANGES_REQUESTED** [review]
Currently just a quick review at a glance, but the "match with a cs student" isn't really applicable here. Originally we were going to do 1:1 video matching for each event, however this has been scrapped and moved towards a single 1:1 video matching queue for all logged in users regardless of the event.

TL;DR could you remove the "match with a cs student" section

**Review by amishshah — CHANGES_REQUESTED** [review]
Also, can we use the standard Material UI buttons? The current buttons you have will look very inconsistent later on.

**Comment by zainja** [comment]
these aren't buttons

**Comment by zainja** [comment]
> Currently just a quick review at a glance, but the "match with a cs student" isn't really applicable here. Originally we were going to do 1:1 video matching for each event, however this has been scrapped and moved towards a single 1:1 video matching queue for all logged in users regardless of the event.
> 
> TL;DR could you remove the "match with a cs student" section

this is just a filler component just to show how any info on that part will look like, im waiting for the exact text to be put there

File: src/components/routes/EventPage.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	useEffect(() => {
+		// expecting API CALL
+		setLoading(true);
+		setName('Hello');
+		setStartDate('2/12');
+		setEndDate('23/12');
+		// setDescription('An Amazing event');
+		setChannelID('eeee');
+		setLoading(false);
+	}, []);
```
Comment (amishshah) [inline src/components/routes/EventPage.tsx]: Could you use a single state object for this? This would be quite a lot of calls to the state and would lead to multiple rerenders unnecessarily.

Also, does this even need to be a stateful component? I don't see event data changing that frequently, this makes more sense as props imo. What do you think @HishamYahya?

File: src/react-app-env.d.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 // / <reference types="react-scripts" />
+declare module '*.jpg';
```
Comment (amishshah) [inline src/react-app-env.d.ts:2]: What is this for?

File: src/App.tsx [hunk 3]
```diff
@@ -21,6 +22,7 @@ function App() {
 						<PublicRoute path="/login" exact component={LoginPage} />
 						<PublicRoute path="/verify" exact component={VerifyEmailPage} />
 						<ProtectedRoute path="/account" exact component={AccountSettingsPage} />
+						<PublicRoute path="/event/:id?" exact component={EventPage}/>
```
Comment (amishshah) [inline src/App.tsx]: This should actually be a ProtectedRoute, users aren't allowed to fetch any form of data from the API unless they are authenticated.

**Review by amishshah — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (zainja) [reply src/App.tsx]: yes i made it public for now because i couldn't view it before

File: src/react-app-env.d.ts (hunk 2 above)
Comment (zainja) [reply src/react-app-env.d.ts:2]: this is to make typescript not flag an image import as an error, react imports the actual image 
import img '../../////'

File: src/components/routes/EventPage.tsx (hunk 1 above)
Comment (zainja) [reply src/components/routes/EventPage.tsx]: im not sure if we want it to be stateful or not because im not sure if you get all the data required from the start

Comment (amishshah) [reply src/components/routes/EventPage.tsx]: Ok, can you still make this a single object for the state though?

File: src/react-app-env.d.ts (hunk 2 above)
Comment (amishshah) [reply src/react-app-env.d.ts:2]: In that case could you add '.png' alongside it!

**Review by amishshah — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -21,6 +22,7 @@ function App() {
 						<PublicRoute path="/login" exact component={LoginPage} />
 						<PublicRoute path="/verify" exact component={VerifyEmailPage} />
 						<ProtectedRoute path="/account" exact component={AccountSettingsPage} />
+						<ProtectedRoute path="/event/:id?" exact component={EventPage}/>
```
Comment (amishshah) [inline src/App.tsx]: IMO this route should be 
```suggestion
						<ProtectedRoute path="/events/:id" exact component={EventPage}/>
```

Since this is convention for specifying a single item of a collection in a HTTP route, the plural `/events` contains all routes, and `/events/:id` indicates a specific event

File: src/components/util/useStyles.ts [hunk 5]
```diff
@@ -26,6 +26,30 @@ const useStyles = makeStyles(theme => ({
...
+	'eventButton': {
+		alignSelf: 'center',
+		margin: 2
+	},
+	'eventOptionText': {
+		margin: theme.spacing(3, 0)
+	},
+	'eventGrid': {
+		margin: theme.spacing(2, 0, 0, 0)
+	},
```
Comment (amishshah) [inline src/components/util/useStyles.ts]: So this `useStyles` file is technical debt from when I first started learning Material UI - we should get rid of it but this will be done in a later PR instead. For now, could you move these new rules you've created into the file they're used in? You can do `makeStyles` in that file. Have a look at the registration page for an example of how to do this.

**Review by amishshah — CHANGES_REQUESTED** [review]
Looks mostly good! Just a few minor changes.

**Review by amishshah — CHANGES_REQUESTED** [review]
I think the page looks visually great! But could you use the `<Page>` component to wrap around it? Since the navbar currently disappears when viewing the page.

**Comment by zainja** [comment]
> I think the page looks visually great! But could you use the `<Page>` component to wrap around it? Since the navbar currently disappears when viewing the page.

done I'll test if the nav bar shows

**Review by amishshah — CHANGES_REQUESTED** [review]
Code looks good, but can you fix lint errors by removing the duplicate store import?

**Review by amishshah — APPROVED** [review]
